RELEASE.md

```diff
@@ -39,54 +39,6 @@ The only uncompleted ticket in the release should be the release ticket.
 If there are any remaining tickets that will not be included in this release, remove the fixVersion and assign them a new one if appropriate.
 Close the release on JIRA, adding the current date.
 
-### Releasing
```

Comment:
The git diff mangled this a bit, but what really happened here is that I moved the "Update the changelog" section into pre-release tasks instead of post-release tasks

Reply:
One issue with that is the "update the changelog" section says the following:

> - Update the release date to the date the release-json task finished on Evergreen in Eastern Time. You can set your timezone in "User Settings". 

I think it could be useful to have an accurate date for when the release actually became available for download. Do we want to edit this date post-release, or use the date when the tag was cut, or remove the date entirely?

Reply:
good point! moved it back to where it was in the process